kbase-extension/static/kbase/js/common/cellComponents/filePathWidget.js

```diff
@@ -48,57 +128,47 @@ define([
-            fieldWidget.bus.on('changed', (message) => {
-                paramsBus.send(
-                    {
-                        parameter: parameterSpec.id,
-                        newValue: message.newValue,
-                        isError: message.isError,
-                    },
-                    {
-                        key: {
-                            type: 'parameter-changed',
-                            parameter: parameterSpec.id,
-                        },
-                    }
-                );
+            /**
+             * We only expect file input and object output parameters here. These are either the
+             * dynamic dropdown widget (for file lookups) or the newObjectInput widget for the
+             * object outputs. Bus messages that get listened to here reflect that - specialized
+             * messages like "sync-params" are only used by a few inputs, none of which are
+             * expected here.
+             */
 
+            // When the field widget gets its value changed (or communicates that it's changed),
+            // put that in the filePathWidget's data model, then push all the values up the main
+            // params bus.
+            fieldWidget.bus.on('changed', (message) => {
+                const rowIndex = dataModel.rowIdToIndex[rowId];
+                dataModel.dataValues[rowIndex][parameterSpec.id] = message.newValue;
                 paramsBus.emit('parameter-changed', {
                     parameter: parameterSpec.id,
                     newValue: message.newValue,
                     isError: message.isError,
+                    rowId: rowId,
+                    rowIndex: dataModel.rowOrder.indexOf(rowId),
                 });
             });
 
-            fieldWidget.bus.on('touched', () => {
-                paramsBus.emit('parameter-touched', {
-                    parameter: parameterSpec.id,
-                });
-            });
-
-            // An input widget may ask for the current model value at any time.
+            // The 'sync' message is a request for the current model value from the
+            // input widget.
             fieldWidget.bus.on('sync', () => {
-                paramsBus.emit('parameter-sync', {
-                    parameter: parameterSpec.id,
-                });
-            });
-
-            fieldWidget.bus.on('sync-params', (message) => {
-                paramsBus.emit('sync-params', {
-                    parameters: message.parameters,
-                    replyToChannel: fieldWidget.bus.channelName,
-                });
-            });
-
-            fieldWidget.bus.on('set-param-state', (message) => {
-                paramsBus.emit('set-param-state', {
-                    id: parameterSpec.id,
-                    state: message.state,
-                });
+                const rowIndex = dataModel.rowIdToIndex[rowId];
+                const value = dataModel.dataValues[rowIndex][parameterSpec.id];
```

Comment:
`value` is already defined in this scope -- see `makeFieldWidget` args

Reply:
Fixed.